### ISSUE TITLE
markFeatureWriter mark class grouping is problematic

### ISSUE THREAD

**Issue by khaledhosny** [googlefonts/ufo2ft#762]
I was debugging a rather odd InDesign bug, where all my mark positioning gets lost, and I tracked it down to:
https://github.com/googlefonts/ufo2ft/blob/7807ef54076238fbe916905bca64a5830566c865/Lib/ufo2ft/featureWriters/markFeatureWriter.py#L456-L501

Usually I’d brush this as an InDesign bug and move on, but I noticed something with HarfBuzz. Try with [this version](https://github.com/aliftype/amiri/raw/1a14935bf34127c1c663e4952ffa738c98d63bfb/fonts/Amiri-Regular.ttf) of the font with mark class grouping:
```
$ hb-view Amiri-Regular.ttf "لَكم محَبر"
```
![before](https://github.com/googlefonts/ufo2ft/assets/93914/b6c22b96-6ff9-4a7f-ae7e-db55d1fee739)

and now with this version without grouping:
![after](https://github.com/googlefonts/ufo2ft/assets/93914/5296026f-b5cb-4014-9dd1-cce29d147e2b)

The later output is the correct/expected one.

The file without grouping is smaller as well despite having more number of lookups (I noticed this with other fonts too), so I’m not sure what is the benefit of this mark class grouping over doing the simple thing?

**Comment by khaledhosny** [comment]
Yes, I get messages about ambiguous anchors, but I get that with pretty much all of my fonts that I build with fontmake.

**Comment by khaledhosny** [comment]
I’m using ufo2ft directly here, though, and I have some hacky code to force a particular lookup order. This probably gets broken when some anchors are merged together in the same lookup (This font has a long history, and for most of its life it was developed using FontForge where I had complete control on how the lookups are built and ordered, which I switched to UFO/ufo2ft I had to hack my way to get a matching output).

**Comment by belluzj** [comment]
Maybe you could use your own copy of the mark feature writer in which you disable grouping?

**Comment by anthrotype** [comment]
maybe grouping should be smarter and avoid grouping ambiguous attachements

**Comment by khaledhosny** [comment]
I’m essentially using my own mark feature writer, it is just this is second time I get issues with this grouping logic (the other one was in a certain secret project, and I ended up forking the mark feature writer there as well, but I don’t remember the exact details of the issue).

I’m also not sure the number of lookups can be that a performance issue, it is not like you are going to end up with thousands of them. This font has too many anchor classes than most typical projects, and I got only 8 more lookups (that is out of 62 GPOS lookups).
